app/models/incomeSourceDetails/IncomeSourceDetailsResponse.scala

```diff
@@ -32,17 +33,65 @@ case class IncomeSourceDetailsModel(mtdbsa:String,
 
   override def toJson: JsValue = Json.toJson(this)
 
+  private def isModelUserMigrated: Boolean = {
+    /**
+     * The function examines content of the yearOfMigration option of the model.
+     * Returns boolean true if this model is for a migrated user, false otherwise
+     */
+    this.yearOfMigration match {
+      case Some(_) => true
+      case _ => false
+    }
+  }
+
+  def firstAccountingTaxYear(yearOption: Option[Int], property: Option[PropertyDetailsModel]): Int = {
+    /**
+     * The function takes list of BusinessDetailsModel and an object of type Option[Int]
+     * Returns option content if any, throws an exception of list of business details models is not empty.
+     * The semantics of the function implements the non-empty first accounting period enforcement in the
+     * presence of the non-empty list of business models involved.
+     * Returns sentinel value of Int.MaxValue only to accommodate existing unit testing only, semantically
+     * meaningless.
+     */
+     if (!this.isModelUserMigrated) {
+       Int.MaxValue
+     } else {
+       yearOption match {
+         case Some(yearInt) => yearInt
+         case _ => {
+           property match {
+             case None => Int.MaxValue
+             case _ => {
+               businesses match {
+                 case Nil => Int.MaxValue
```

Comment:
this would cause no exception to be thrown for some one who had no business but some property income, this is valid use case and would have to throw the missing accounting period exception